pkg/distributor/distributor.go

```diff
@@ -189,21 +191,25 @@ func (d *Distributor) Push(ctx context.Context, req *logproto.PushRequest) (*log
 	// We also work out the hash value at the same time.
 	streams := make([]streamTracker, 0, len(req.Streams))
 	keys := make([]uint32, 0, len(req.Streams))
-	var validationErr error
+	errSampler := util.NewCodedErrorSampler(5)
 	validatedSamplesSize := 0
 
 	for _, stream := range req.Streams {
 		if err := d.validateLabels(userID, stream.Labels); err != nil {
-			validationErr = err
+			if !errSampler.IsFull(err) {
```

Comment:
Is there a reason `AddEntry` can't just internally call `IsFull`?

Reply:
Agree !

Reply:
I think we should have this for performance reasons, this allows saving the allocations for the error message on the following line, there is no reason to generate the error message string if we aren't going to keep it